### PR TITLE
Use firmware.upgrade

### PR DESCRIPTION
**Pull request by kasperl**

### CODE REVIEW

File: src/jaguar.toit [hunk 1]
```diff
@@ -53,10 +53,11 @@ registry_ / ContainerRegistry ::= ContainerRegistry
...
   try:
     catch --trace: registry_.start_installed
-    exception := catch --trace: serve arguments
-    logger.error "rebooting due to $(exception)"
-  finally:
-    esp32.deep_sleep (Duration --s=1)
+    serve arguments
+    unreachable
+  finally: | is_exception exception |
+    cause := is_exception ? exception.value : null
```
Comment (floitsch) [inline src/jaguar.toit:59]: Do we have something better than "null" here?
"rebooting due to null" looks like a bug.

Is it even possible to not be in an exception?

File: src/jaguar.toit [hunk 2]
```diff
@@ -53,10 +53,11 @@ registry_ / ContainerRegistry ::= ContainerRegistry
...
+    logger.error "rebooting due to $cause"
```
Comment (floitsch) [inline src/jaguar.toit:60]: Is there an automatic reboot when leaving main?

**Review by floitsch — APPROVED** [review]
LGTM.

File: src/jaguar.toit (hunk 1 above)
Comment (kasperl) [reply src/jaguar.toit:59]: Added comment and improved code a bit.

File: src/jaguar.toit (hunk 2 above)
Comment (kasperl) [reply src/jaguar.toit:60]: Added comment.

**Pull request merged**